cluster_test.go

```diff
@@ -1671,6 +1671,37 @@ func (s *S) TestMaxSocketUseTimeExpireAfterRelease(c *C) {
 	c.Assert(stats.SocketsExpired, Equals, 1)
 }
 
+// test to ensure that expired sockets are removed from active sockets list
+func (s *S) TestSocketExpiryEnsureSocketClosed(c *C) {
```

Comment:
I had to stare at this for a long time before I understood what is was trying to do, and I'm still a bit fuzzy.

As far as I can tell we are:

* Exploiting the pool size maximum of 1
* Causing a session to be created and then held as "used" indefinitely
* Causing background goroutine to try to obtain another one
* Rely on the expiry of the existing session happening during a sleep (vs. only happening when use of it is attempted), therefor allowing the test not to deadlock.
* Finally, measure the time the background goroutine blocked expecting it to be high enough to prove that it had to wait for expiry.

The part I'm still fuzzy on is how the expiry happens. I can probably figure it out by revisiting the previous PR or the code. Perhaps a comment on top along the lines of:

```
Create a pool with a maximum pool size of 1, and try to use 2 concurrent sessions. Measure that the
second session's acquisition (in the background goroutine) blocks waiting for the first session to be expired, thus showing we were not able to keep two concurrent sessions open violating the pool size.
```

I'm really fuzzy on whether the above is correct.